Update VS Code extension docs to cover Code Insights and View in VS Code (#18770)
DataDog/documentation: content/en/developers/ide_integrations/vscode/_index.md

---

> title: Datadog Extension for VS Code

Reviewer: ```suggestion
title: Datadog Extension for Visual Studio Code
```

- link: "/continuous_testing/testing_tunnel/"
  tag: "Documentation"
  text: "Learn about the Continuous Testing Tunnel"
- link: "/integrations/guide/source-code-integration/"
  tag: "Documentation"
  text: "Learn about Source Code Integration."
---

{{< callout url="#" btn_hidden="true">}}
  The Datadog extension for VS Code is in public beta. It is intended for all developers that use Datadog products, and especially those using <a href="https://docs.datadoghq.com/synthetics/#pagetitle">Synthetic tests</a>. If the extension stops working unexpectedly, check for extension updates or <a href=#feedback>reach out to the team</a>.
{{< /callout >}}

> ## Overview

Reviewer 2: We have a section for `View in VS Code` but there's no mention of it in the overview section.

Author: Added

> The Datadog extension for VS Code integrates with the Datadog platform to accelerate your development.

Reviewer: ```suggestion
The Datadog extension for VS Code integrates with Datadog to accelerate your development.
```

{{< img src="/developers/ide_integrations/vscode/datadog-vscode.png" alt="The Datadog for VS Code extension" style="width:100%;" >}}

The **Code Insights** view keeps you informed about:
- Issues from [Error Tracking][10]
- [Vulnerability][11] reports by Application Security Management
- [Flaky tests][12] detected by CI Visibility
- Profiling insights from [Watchdog Insights][13]

The **Synthetic Tests** feature allows you to:
- Run Synthetic tests on your local environments
- Set custom parameters without altering the original test definition
- See test results locally in VS Code and in Datadog to access additional information
- Test only what matters by executing relevant tests at the same time

> - Create a list of most frequently used Synthetic tests by adding them to **Favorites** 

Reviewer 2: Not sure if I would use the word `Favorites` here as we don't have this concept in the IDE anymore.

Author: OK, changing to `by linking them to your workspace`

- link: "/continuous_testing/testing_tunnel/"
  tag: "Documentation"
  text: "Learn about the Continuous Testing Tunnel"
- link: "/integrations/guide/source-code-integration/"
  tag: "Documentation"
  text: "Learn about Source Code Integration."
---

{{< callout url="#" btn_hidden="true">}}
  The Datadog extension for VS Code is in public beta. It is intended for all developers that use Datadog products, and especially those using <a href="https://docs.datadoghq.com/synthetics/#pagetitle">Synthetic tests</a>. If the extension stops working unexpectedly, check for extension updates or <a href=#feedback>reach out to the team</a>.
{{< /callout >}}

## Overview

> The Datadog extension for VS Code integrates with Datadog to accelerate your development.

Reviewer: ```suggestion
The Datadog extension for Visual Studio Code (VS Code) integrates with Datadog to accelerate your development.
```

Install the [Datadog Extension][6] from the Visual Studio Marketplace.

> ## Code Insights

Reviewer 2: I put the comment here as I can't comment in the section itself but this comment is related to the `Requirements` section. I would probably refactor that a little bit considering that Synthetic tests are not required for the extension to work, and neither having a Datadog Account if we only consider View in VS Code. WDYT?

Author: View in VS Code doesn't **strictly** require an account, but the main use-case is clicking links on Datadog and for that you do need an account, so I think this is OK. 

I see your point about the Synthetic Tests - I moved that text to the section about running tests locally.

The **Code Insights** tree displays insights generated by the Datadog platform that are relevant to your code-base. The insights are grouped into three categories: performance, reliability, and security.

{{< img src="/developers/ide_integrations/vscode/code-insights.png" alt="The Code Insights view." style="width:100%;" >}}

Code Insights include a detailed description for each issue, and links to:
- The related source code location
- The Datadog platform for additional information

> Developers can dismiss individual insights and set filters to view the categories of insights that they are most interested to see.

Reviewer: ```suggestion
You can dismiss individual insights and set filters to view the categories of insights that you are most interested to see.
```

## Synthetic Tests

> The Datadog extension enables you to [run Synthetic HTTP tests and browser tests on local environments][1] directly in the IDE. This ensures that any potential issue resulting from code changes is identified and addressed before the code is deployed into production and impacts your end users.

Reviewer: ```suggestion
The Datadog extension enables you to [run Synthetic HTTP tests and browser tests on local environments][1] directly in the IDE. You can identify and address potential issues resulting from code changes before they are deployed into production and impact your end users.
```

{{< img src="developers/ide_integrations/vscode/vscode-extension-demo.png" alt="The Datadog Extension in VS Code" style="width:100%;" >}}

### Run Synthetic tests locally

1. Select a Synthetic test to execute. You can search for specific tests by clicking the **Search** icon.

> 2. To run a test in a local environment, update a Synthetic test's configuration by enabling the conversion of the start URL and specifying a `localhost` URL on the **Settings** page.

Reviewer: ```suggestion
2. Change the test's configuration to convert the start URL and specify a `localhost` URL on the **Settings** page.
```

> 3. Run a test.

Reviewer: ```suggestion
3. Run the test.
```

{{< img src="developers/ide_integrations/vscode/test_configuration_modified_starturl.png" alt="The Test Configuration panel and Settings page where you can specify the start URL of a Synthetics test to a localhost URL" style="width:100%;" >}}

### Permissions

By default, only users with the [Datadog Admin and Datadog Standard roles][7] can create, edit, and delete Synthetic HTTP and browser tests. To get create, edit, and delete access to Synthetic HTTP and browser tests, upgrade your user to one of those two [default roles][7].

If you are using the [custom role feature][8], add your user to any custom role that includes `synthetics_read` and `synthetics_write` permissions.

## View in VS Code

> The **View in VS Code** feature provides a link from the Datadog platform directly to your source files. Look for the button next to frames in stack traces displayed on the platform (for example, in [Error Tracking][10]):

Reviewer: ```suggestion
The **View in VS Code** feature provides a link from Datadog directly to your source files. Look for the button next to frames in stack traces displayed in the UI (for example, in [Error Tracking][10]):
```

{{< img src="/developers/ide_integrations/vscode/view-in-vscode.png" alt="A stack trace on the Datadog platform showing the View in VS Code button." style="width:100%;" >}}

> <div class="alert alert-info">A prerequisite for this feature is that Source Code Integration is configured for your service.</div>

Reviewer: ```suggestion
<div class="alert alert-info">To use this feature, first configure <a href="/integrations/guide/source-code-integration/">source code integration</a> for your service.</div>
```
